src/client/components/leaderboard/leaderboard.js

```diff
@@ -10,6 +11,14 @@ class Leaderboard extends React.Component {
     playerID: PropTypes.any.isRequired,
   };
   render() {
+    function haspassed(_idx) {
```

Comment:
Could you use camel case, i.e. `hasPassed`, please?

Reply:
Generally, it is the convention that a function called `hasSomething` or `isSomething` returns a boolean value. Maybe you could rename it to something like `getPassedMessage` or so...

Reply:
I believe you could write this simpler as 
```js
function getPassedMessage(_idx) {
  return passedexp.some(
    passedPlayerId => passedPlayerId === _idx.toString()
  );
}
```